keyboard/src/modules/Keyboard.rb

```diff
@@ -1030,40 +1030,20 @@ def SetConsole(keyboard)
     #		(also stored in Keyboard::xkb_cmd)
     def SetX11(keyboard)
       if Mode.test
-        Builtins.y2milestone("Test mode - would have called:\n %1", @xkb_cmd)
+        log.info "Test mode - would have called:\n #{@xkb_cmd}"
       else
         # Actually do it only if we are in graphical mode.
         #
-        textmode = Linuxrc.text
-        if !Stage.initial || Mode.live_installation
-          display_info = UI.GetDisplayInfo
-          textmode = Ops.get_boolean(display_info, "TextMode", false)
-        end
-        display = Builtins.getenv("DISPLAY")
-        if textmode
-          Builtins.y2milestone("Not setting X keyboard due to text mode")
+        if textmode?
+          log.info "Not setting X keyboard due to text mode"
         # check if we are running over ssh: bnc#539218,c4
-        elsif Ops.greater_or_equal(
-            Builtins.tointeger(
-              Ops.get(Builtins.splitstring(display, ":"), 1, "0")
-            ),
-            10
-          )
-          Builtins.y2milestone("Not setting X keyboard: running over ssh")
-        elsif Ops.greater_than(Builtins.size(@xkb_cmd), 0)
+        elsif x11_over_ssh?
```

Comment:
Yeah, actually methods like ```textmode?``` or ```x11_over_ssh?``` is the kind of things that you would not expect to have to implement in every module/client. Although I suspect that all the implementations of similar checks we have spread all over the YaST code have slightly different semantic from each other, so not so easy to unify.